modules/ROOT/pages/supported-types-of-tests.adoc

```diff
@@ -0,0 +1,41 @@
+= MUnit Supported Types of Tests
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+The following are definitions and concepts that MuleSoft recommends to take into account while designing and organizing test cases for your Mule applications. Understanding these definitions and concepts will help you build more robust MUnit tests by covering multiple edge cases, automate your tests through your CI/CD pipeline, and maximize the quality coverage for your Mule application.
+
+As a general good practice, MuleSoft recommends following a naming convention for file names and tests so that you can quickly realize what was being tested when a test fails.
+
+When testing a Mule application MUnit allows you to cover three different types of tests:
+
+* Unit tests.
+* Integration tests.
+* Functional test.
+
+== Unit Testing
+
+Unit testing consists of testing individual units or components of a program. A unit is the smallest testable part of any software. For a Mule application, these units are flows and when doing unit testing for flows, you must isolate them from the rest of the Mule application or external services by mocking these calls to any resource outside the flow you are testing.
+
+Creating these tests helps you validate the behavior of key processors and the behavior of a flow as a whole.
+
+== Integration testing
+
+When running integration tests you must validate how the units interact with each other or with external systems to expose faults in the interactions. Samples of integration testing in a Mule application are:
+
+* Testing how a service responds (Web Services, Salesforce, ServiceNow, Amazon, etc).
+* Testing a database driver.
+* Validating a database response data structure.
+* Testing in-between flows interactions.
```

Comment:
Does this mean testing the interactions between stages of a flow?
Are the parts of flows called "stages"?
Will the audience wonder if the smallest testable unit is the stage? Prolly not....